### PR TITLE
DISPATCH-1096 - priority sheaf cleanup bug

### PR DESCRIPTION
**Pull request by mgoulish**
This is for the bug that Ganesh and Chuck saw while killing and restarting routers.
The test succeeded 10 times each (with 0 false results) in these 4 cases:
   { Debug, RelWithDebInfo } x { Should_Fail, Should_Succeed }

### CODE REVIEW

**Comment by ChugR** [comment]
This patch clears up the too-many-links for sheaf error and improves my router network stability. FOr this I approve the patch.

But in tests before and after this patch my throughput went from 2600 msg/S to 200-300 msg/S. The performance drop needs to be addressed as a separate issue.

File: src/router_core/connections.c [hunk 1]
```diff
@@ -1466,8 +1470,8 @@ static void qdr_attach_link_data_CT(qdr_core_t *core, qdr_connection_t *conn, qd
         // are assigned priorities in the order in which they are attached.
         int next_slot = core->data_links_by_mask_bit[conn->mask_bit].count ++;
         if (next_slot >= QDR_N_PRIORITIES) {
-            qd_log(core->log, QD_LOG_ERROR, "Attempt to attach too many inter-router links for priority sheaf.");
-            return;
+            qd_log(core->log, QD_LOG_CRITICAL, "Attempt to attach too many inter-router links for priority sheaf.");
+            exit(1);
```
Comment (ted-ross) [inline src/router_core/connections.c]: Exiting the process here is a bit heavy handed.  I think it's sufficient to log the critical error.

**Comment by ted-ross** [comment]
> This patch clears up the too-many-links for sheaf error and improves my router network stability. FOr this I approve the patch.
> 
> But in tests before and after this patch my throughput went from 2600 msg/S to 200-300 msg/S. The performance drop needs to be addressed as a separate issue.

I don't see any changes in this PR that should affect throughput performance.

**Comment by ted-ross** [comment]
> > This patch clears up the too-many-links for sheaf error and improves my router network stability. FOr this I approve the patch.
> > But in tests before and after this patch my throughput went from 2600 msg/S to 200-300 msg/S. The performance drop needs to be addressed as a separate issue.
> 
> I don't see any changes in this PR that should affect throughput performance.

I withdraw the above statement.  There appears to be a memory leak that is causing performance (and other) problems.

**Comment by mgoulish** [comment]
I yanked that exit(1).
I thought I needed that to detect failure in my python test, but that is not true.
Tests all still did the right thing, in 10 trials each (40 total) in the set:    {should_fail, should_succeed} *  { DEBUG, RelWithDebInfo}

**Pull request closed without merging**